metadata-ingestion/src/datahub/ingestion/source/redash.py

```diff
@@ -639,11 +656,12 @@ def _emit_chart_mces(self) -> Iterable[MetadataWorkUnit]:
             for query_response in queries_response["results"]:
 
                 chart_name = query_response["name"]
-
+                database_name = self._get_database_name_from_datasource_id(query_response["data_source_id"])
                 self.report.report_item_scanned()
 
                 if (not self.config.chart_patterns.allowed(chart_name)) or (
-                    skip_draft and query_response["is_draft"]
+                    skip_draft and query_response["is_draft"]) or (
+                    self.config.database_patterns.allowed(database_name)
```

Comment:
Don't you want to check if the database name is NOT allowed? 

ie. `not self.config.database_patterns.allowed(database_name)`

Reply:
Confused about why this case is here - I see the other skip case above

Reply:
That's true. I'll check every if sentences and rewrite them. But Yes, i should check for not self.config... Thanks for the review!

Reply:
Now related to the case here it's because it's 2 different requests to the redash api, so i need to check the allowed databases from the both requests

Reply:
solved at: https://github.com/datahub-project/datahub/pull/4616/commits/b1794247f7569e52b27e2107905413444db00b68